test/spec.model.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const SessionModel = require('../lib/model');
+
+describe('session model', () => {
+
+    it('exports a function', () => {
+        SessionModel.should.be.a.function;
+    });
+
+    describe('constructor', () => {
+        it('should throw an error if no options are provided', () => {
+            expect(() => new SessionModel()).to.throw('session-model - key must be defined');
+        });
+
+        it('should throw an error if no key is provided in options', () => {
+            expect(() => new SessionModel(null, { session: {} })).to.throw('session-model - key must be defined');
+        });
```

Comment:
Have updated to test for errors when the key is not a string and when the session is not an object :+1: